### PR TITLE
fix(spantest): cloudbuild host address

### PR DESCRIPTION
**Pull request by kristofferwanglund**
The cloudbuild spanner test instance needs
docker inspect to get the host address.

### CODE REVIEW

File: tools/sgcloudspanner/emulator.go [hunk 1]
```diff
@@ -65,10 +66,20 @@ func RunEmulator(ctx context.Context) (_ func(), err error) {
 			sg.Logger(ctx).Printf("failed to unset SPANNER_EMULATOR_HOST: %v", err)
 		}
 	}
-	emulatorHost, err := inspectPortAddress(ctx, containerID, "9010/tcp")
-	if err != nil {
-		cleanup()
-		return nil, err
+
```
Comment (odsod) [inline tools/sgcloudspanner/emulator.go:69]: IMO remove this blank line

File: tools/sgcloudspanner/emulator.go [hunk 2]
```diff
@@ -107,6 +118,48 @@ func inspectPortAddress(ctx context.Context, containerID, containerPort string)
...
+			}
+			Networks map[string]struct {
+				Gateway string
+			}
+		}
+	}
+	if err := json.NewDecoder(&stdout).Decode(&containers); err != nil {
+		return "", err
+	}
+	fmt.Printf("containers:\n%+v", containers)
```
Comment (odsod) [inline tools/sgcloudspanner/emulator.go:145]: please remove

File: tools/sgcloudspanner/emulator.go [hunk 3]
```diff
@@ -107,6 +118,48 @@ func inspectPortAddress(ctx context.Context, containerID, containerPort string)
...
+	var host string
```
Comment (odsod) [inline tools/sgcloudspanner/emulator.go:146]: `var host, port string`

File: tools/sgcloudspanner/emulator.go [hunk 4]
```diff
@@ -107,6 +118,48 @@ func inspectPortAddress(ctx context.Context, containerID, containerPort string)
...
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+	var containers []struct {
+		Config struct {
+			Image string
+		}
+		NetworkSettings struct {
+			Ports map[string][]struct {
+				HostIP   string
```
Comment (thall) [inline tools/sgcloudspanner/emulator.go:134]: is this used?

File: tools/sgcloudspanner/emulator.go (hunk 2 above)
Comment (odsod) [reply tools/sgcloudspanner/emulator.go:145]: or replace with proper logging using `sg.Logger`

File: tools/sgcloudspanner/emulator.go [hunk 5]
```diff
@@ -107,6 +118,48 @@ func inspectPortAddress(ctx context.Context, containerID, containerPort string)
...
+	}
+	fmt.Printf("containers:\n%+v", containers)
+	var host string
+	var port string
+
+	for _, container := range containers {
+		if container.Config.Image == image {
+			port = container.NetworkSettings.Ports[containerPort][0].HostPort // prefer first option
+		}
+		if network, ok := container.NetworkSettings.Networks["cloudbuild"]; ok {
```
Comment (thall) [inline tools/sgcloudspanner/emulator.go:153]: can we break out `cloudbuild` to a constant? used in three places.

File: tools/sgcloudspanner/emulator.go (hunk 4 above)
Comment (kristofferwanglund) [reply tools/sgcloudspanner/emulator.go:134]: No it should not be. Removing.

File: tools/sgcloudspanner/emulator.go (hunk 5 above)
Comment (kristofferwanglund) [reply tools/sgcloudspanner/emulator.go]: Yes, done.

File: tools/sgcloudspanner/emulator.go (hunk 3 above)
Comment (kristofferwanglund) [reply tools/sgcloudspanner/emulator.go]: Fixed.

File: tools/sgcloudspanner/emulator.go (hunk 2 above)
Comment (kristofferwanglund) [reply tools/sgcloudspanner/emulator.go]: Oops, that should not have been there. Removed.

File: tools/sgcloudspanner/emulator.go (hunk 1 above)
Comment (kristofferwanglund) [reply tools/sgcloudspanner/emulator.go]: Done.

**Review by odsod — APPROVED** [review]
Good stuff!

**Review by thall — APPROVED** [review]
Nice!

**Pull request merged**